starknet_py/net/schemas/rpc.py

```diff
@@ -230,17 +231,17 @@ def make_dataclass(self, data, **kwargs) -> ResourceBoundsMapping:
 
 
 class TransactionSchema(Schema):
-    hash = Felt(data_key="transaction_hash", load_default=None)
-    signature = fields.List(Felt(), data_key="signature", load_default=[])
-    version = Felt(data_key="version", required=True)
-
-
-class DeprecatedTransactionSchema(TransactionSchema):
-    max_fee = Felt(data_key="max_fee", required=True)
+    type = fields.Str(required=True, validate=validate.OneOf(["DECLARE"]))
+    sender_address = Felt(data_key="sender_address", required=True)
+    signature = fields.List(Felt(), data_key="signature", required=True)
+    class_hash = Felt(data_key="class_hash", required=True)
 
 
-class TransactionV3Schema(TransactionSchema):
```

Comment:
I would definitely prefer to keep `TransactionV3Schema` to group common V3 transaction fields together. How about we keep things simple for now and think of enhancements later? So, here's the inheritance structure:
- `TransactionSchema` -> `DeprecatedTransactionSchema` ->:
  - `InvokeTransactionV0Schema`, `InvokeTransactionV1Schema`
  - `DeclareTransactionV0Schema`, `DeclareTransactionV1Schema` and `DeclareTransactionV2Schema`
  - `DeployAccountTransactionV1Schema`
- `TransactionSchema` -> `TransactionV3Schema` -> all V3 transaction schemas (no changes)

Please pay attention to the `required` property for all fields across different schemas.